src/GameStateMenu.cpp

```diff
@@ -10,143 +10,509 @@
+            if (keyCode == SDLK_ESCAPE) {
                 play = false;
                 break;
-            default:
-                break;
             }
-            break;
-          case SDL_KEYUP:
-               switch( event.key.keysym.sym ) {
-            default:
-                   break;
+
+            for (size_t i = 0; i < NUM_TEXT_FIELDS; i++) {
+                if (activeTextbox[i]) {
+
+                    //Handling backspace
+                    if (keyCode == SDLK_BACKSPACE && textInput[i].length() > 0) {
+                        textInput[i].pop_back();
+                    }
+
+                    //Handle copy
+                    else if (keyCode == SDLK_c && SDL_GetModState() & KMOD_CTRL) {
+                        SDL_SetClipboardText( textInput[i].c_str() );
+                    }
+
+                    //Handle paste
+                    else if (keyCode == SDLK_v && SDL_GetModState() & KMOD_CTRL) {
+                        textInput[i] = SDL_GetClipboardText();
+                    }
+                }
+            }
+        	break;
+        case SDL_TEXTINPUT:
+            for (size_t i = 0; i < NUM_TEXT_FIELDS; i++) {
+                if (activeTextbox[i]) {
+                    //Ensures that a copy or paste is not occurring
+                    if (!((event.text.text[ 0 ] == 'c' || event.text.text[ 0 ] == 'C') &&
+                          (event.text.text[ 0 ] == 'v' || event.text.text[ 0 ] == 'V') &&
+                           SDL_GetModState() & KMOD_CTRL)) {
+                        //Update the string int displayed int he textbox
+                        if (textInput[i].length() < maxLength) {
+                            textInput[i] += event.text.text;
+                        }
+                    }
+                }
             }
             break;
-        case SDL_QUIT:
-            play = false;
-            break;
-          default:
-            break;
-        }
-    }
+      	case SDL_KEYUP: //Do nothing on key release
+       		switch ( event.key.keysym.sym ) {
+				default:
+	               	break;
+			}
+        	break;
+        case SDL_MOUSEMOTION:
+        	x = event.motion.x;
+        	y = event.motion.y;
+        	for (size_t i = 0; i < NUM_MENU_ITEMS; i++) {
+                //Check if mouse is above one of the menu uptions
+        		if (x >= menuItemPos[i].x && x <= menuItemPos[i].x + menuTextTextures[i].getWidth() &&
+        			y >= menuItemPos[i].y && y <= menuItemPos[i].y + menuTextTextures[i].getHeight()) {
+                    selected[i] = true;  //Activate the button
+        		} else {
+                    selected[i] = false; //deactivate the button
+        		}
+        	}
+        	break;
+        case SDL_MOUSEBUTTONDOWN:
+        	x = event.button.x;
+        	y = event.button.y;
+        	for (size_t i = 0; i < NUM_MENU_ITEMS; i++) {
+                if (selected[i]) {
+                    update(i);
+        			play = false;
```

Comment:
just noticed that this is tabs, please remove them.
for a quick solution there is a simple command [on the wiki](https://github.com/bcit-datacomm/4981_Linux_Game/wiki/Style-Guide#dont-use-tabs-ever-period-use-4-spaces) that you can run